rclpy/test/test_clock.py

```diff
@@ -12,18 +12,23 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import threading
```

Comment:
Some of my comments in the tests cases of https://github.com/ros2/rclpy/pull/864 also apply here

Reply:
Of my comments in the other PR, I think that the only one relevant for the moment is using `thread.join()` instead of a sleep() call to wait the thread to exit.
The other comments can be ignored.